source/DSCResources/DSC_CMClientSettings/DSC_CMClientSettings.psm1

```diff
@@ -0,0 +1,450 @@
+        [String[]]
+        $SecurityScopesToInclude,
+
+        [Parameter()]
+        [String[]]
+        $SecurityScopesToExclude,
+
+        [Parameter()]
+        [ValidateSet('Present','Absent')]
+        [String]
+        $Ensure = 'Present'
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+    $state = Get-TargetResource -SiteCode $SiteCode -ClientSettingName $ClientSettingName -Type $Type
+
+    try
+    {
+        if ($state.Ensure -ne 'Absent' -and $state.Type -ne $Type)
+        {
+            throw ($script:localizedData.TypeMisMatch -f $ClientSettingName)
+        }
+        elseif ($Ensure -eq 'Present')
+        {
+            if ($PSBoundParameters.ContainsKey('SecurityScopes'))
+            {
+                if ($PSBoundParameters.ContainsKey('SecurityScopesToInclude') -or
+                    $PSBoundParameters.ContainsKey('SecurityScopesToExclude'))
+                {
+                    Write-Warning -Message $script:localizedData.ParamIgnoreScopes
+                }
+            }
+            elseif (-not $PSBoundParameters.ContainsKey('SecurityScopes') -and
+                $PSBoundParameters.ContainsKey('SecurityScopesToInclude') -and
+                $PSBoundParameters.ContainsKey('SecurityScopesToExclude'))
+            {
+                foreach ($item in $SecurityScopesToInclude)
+                {
+                    if ($SecurityScopesToExclude -contains $item)
+                    {
+                        throw ($script:localizedData.ScopeInEx -f $item)
+                    }
+                }
+            }
+            elseif (-not $PSBoundParameters.ContainsKey('SecurityScopes') -and
+                -not $PSBoundParameters.ContainsKey('SecurityScopesToInclude') -and
+                $PSBoundParameters.ContainsKey('SecurityScopesToExclude'))
+            {
+                if ($state.SecurityScopes.Count -eq $SecurityScopesToExclude.Count)
+                {
+                    $excludeAll = Compare-Object -ReferenceObject $state.SecurityScopes -DifferenceObject $SecurityScopesToExclude
+                    if ([string]::IsNullOrEmpty($excludeAll))
+                    {
+                        throw ($script:localizedData.ScopeExcludeAll)
+                    }
+                }
+            }
+
+            if ($state.Ensure -eq 'Absent')
+            {
+                if ($PSBoundParameters.ContainsKey('Description'))
+                {
+                    $newParams = @{
+                        Name        = $ClientSettingName
+                        Type        = $Type
+                        Description = $Description
+                    }
+                }
+                else
+                {
+                    $newParams = @{
+                        Name = $ClientSettingName
+                        Type = $Type
+                    }
+                }
+
+                New-CMClientSetting @newParams
```

Comment:
In testing, it looks like the New-CmClientSetting command will by default add the 'Default' scope. May need to add some logic here to Set-CMClientSetting removing the default security scope if the parameters speficy the exclusion of that security scope.